rust/agama-manager/src/start.rs

```diff
@@ -51,8 +53,12 @@ pub async fn start(
     let progress = progress::start(events.clone()).await?;
     let l10n = l10n::start(issues.clone(), events.clone()).await?;
     let storage = storage::start(progress.clone(), issues.clone(), events.clone(), dbus).await?;
+    let network_adapter = network::NetworkManagerAdapter::from_system()
+        .await
+        .expect("Could not connect to NetworkManager");
+    let network = network::NetworkSystem::new(network_adapter).start().await?;
```

Comment:
As you mentioned, perhaps we could have a `NetworkSystem::for_network_manager` so you do no need to worry about the adapter here.